planning/MartenV4-WastazComments.md

```diff
@@ -0,0 +1,174 @@
+# Projections
+
+## Building projections
+While its nice that Marten provides several options for actually constructing projections (`ITransform`, `AggregateStreamWith`, `ViewProjection`, 
+raw `IProjection`, etc...honestly Ive probably forgotten some way of doing it) in the end the ones that I end up using 99% of the time is
+either `ViewProjection` or implementing the raw `IProjection`. The main reason for this is that most of the other options are too limiting
+when building projections. 
+
+Working with `ViewProjection` is quite nice, however occasionally the number of overloads can make it hard because its absolutely not hard
+to make the compiler confused over which overload you are actually trying to use. The main thing that I cant do with `ViewProjection` which
+often is the thing that makes me drop down to `IProjection` is project an event to multiple documents, which is something that can be very useful
+for solving some problems. 
+
+Working with `IProjection` often feels very raw, and I have to admit that I still havent figured out the actual difference between `Apply` and `ApplyAsync`
+because it seems like they are...executed depending on if the projection runs inline or via the async daemon? Thats just confusing tbh.
+
+Also (and now we are getting ahead of myself), projections depend on mutating state. I think this is fine, even fine as a default. But I do wish that
+there would be an option to return an entire new object instead of mutating. This would also help a lot in the F# world.
+
+## Running projections
+Its great that Marten supports both running the projections inline and async. While Im suspecting that some really hardcore cqrs/es people
+balk at the prospect of inline projections, they are a great option to have in some cases even though I try hard not to overuse them.
+
+The async daemon however, is a bit lacking (though it does its job ok). The documentation is an old blog post by Jeremy which doesnt say
+that much about it tbh. And you cant shard or put up multiples of it for some sort of HA scenario. It feels a bit scary just running a single service
```

Comment:
Yes, it's not ideal where it might fail, however, it shouldn't be super hard to configure the Hosted Service (.NET gives an option to configure native services via that) and restart it. It's not ideal, but getting into the distributed zone would make things much harder. The async daemon should catch up the projections and no events should be lost as we're using the UoW there.
I'm not so optimistic as @jeremydmiller that we'll have soon buletproof multi nodes solution for that (as I also stated in the other comment).